### PR TITLE
Apply patches to support upgrading Zepto monolith to active_interaction v4.1

### PR DESCRIPTION
**Pull request by kieranjogrady**
## 💡 Why
Ensures compatibility with v4.1 of `active_interaction` gem.

## 📝 What
Apply [patches](https://github.com/antulik/active_interaction-extras/pulls/kieranjogrady) that have been submitted to upstream project for review. The intention is to require this branch of our fork in our monolith's Gemfile until the upstream PRs are merged.

### CODE REVIEW

**Review by zubin — APPROVED** [review]
Lacking context around the breaking changes but code looks decent.

File: lib/active_interaction/extras/strong_params.rb [hunk 1]
```diff
@@ -5,6 +5,8 @@ def initialize(inputs = {})
     # TODO: whitelist :params and :form_params, so they could not be used as filters
     return super if self.class.filters.key?(:params) || self.class.filters.key?(:form_params)
 
+    return super(inputs) unless %i[key? fetch merge].all? { |m| inputs.respond_to? m }
```
Comment (zachingle) [inline lib/active_interaction/extras/strong_params.rb]: Do we need to explicitly pass `inputs` to `super` here?

**Review by zachingle — APPROVED** [review]
Apart from a small nitpick, LGTM!

File: lib/active_interaction/extras/strong_params.rb (hunk 1 above)
Comment (kieranjogrady) [reply lib/active_interaction/extras/strong_params.rb]: Nope, `super` on its own should suffice. The `super` call below on line 27 passes `inputs` explicitly and I followed that. I suppose we could probably combine this conditional early return with the one above. WDYT?
```
    return super if 
      self.class.filters.key?(:params) ||
      self.class.filters.key?(:form_params) ||
      !%i[key? fetch merge].all? { |m| inputs.respond_to? m }
```

Comment (zachingle) [reply lib/active_interaction/extras/strong_params.rb]: Seems good to me!

**Review by chrisixion — APPROVED** [review]
Looks good, thanks for tackling the prickly issues 🙏 

Caveat - would be nice to roll out in a controlled manor (i.e. bootboot as you proposed)

**Comment by kieranjogrady** [comment]
Closing this PR as a new version has been released for the forked repository that addresses our issues.

**Pull request closed without merging**